File: 예제/ServiceExample/app/src/main/java/com/example/serviceexample/MainActivity.java
package com.example.serviceexample;

import androidx.appcompat.app.AppCompatActivity;

import android.content.Intent;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.widget.Button;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        Button b1 = (Button) findViewById(R.id.button1);
        Button b2 = (Button) findViewById(R.id.button2);

        b1.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                Log.d("test", "액티비티-서비스 시작버튼클릭");
                Intent intent = new Intent(
                        getApplicationContext(),//현재제어권자
                        MyService.class); // 이동할 컴포넌트
                startService(intent); // 서비스 시작
            }
        });

        b2.setOnClickListener(new View.OnClickListener() {
            public void onClick(View v) {
                // 서비스 종료하기
                Log.d("test", "액티비티-서비스 종료버튼클릭");
                Intent intent = new Intent(
                        getApplicationContext(),//현재제어권자
                        MyService.class); // 이동할 컴포넌트
                stopService(intent); // 서비스 종료
            }
        });
    }
}

File: 예제/BottomNavigation/app/src/main/java/com/example/bottomnavigation/MainActivity.kt
package com.example.bottomnavigation

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentTransaction
import com.google.android.material.bottomnavigation.BottomNavigationView
import com.google.android.material.internal.NavigationMenuItemView
import com.google.android.material.navigation.NavigationView

class MainActivity : AppCompatActivity() {

    private fun replaceFragment(fragment: Fragment) {
        val fragmentTransaction: FragmentTransaction = supportFragmentManager.beginTransaction()
        fragmentTransaction.replace(R.id.MyFragment, fragment)
        fragmentTransaction.commit()
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        replaceFragment(FirstFragment())
        replaceFragment(SecondFragment())
        replaceFragment(ThirdFragment())


        val mOnNavigationiItemSelectedListener = BottomNavigationView.OnNavigationItemSelectedListener { Item ->

            when (Item.itemId) {
                R.id.action_first-> {
                    replaceFragment(FirstFragment())
                    return @mOnNavigationiItemSelectedListener true
                }
                R.id.action_second -> {
                    replaceFragment(SecondFragment())
                    return@mOnNavigationiItemSelectedListener true
                }
                R.id.action_third-> {
                    replaceFragment(ThirdFragment())
                    return@mOnNavigationiItemSelectedListener true
                }
                else -> {
                    return@mOnNavigationiItemSelectedListener false
                }

            }
        }
    }
}

    //화면 이동
    //해당 네비게이션을 선택했을 때 해당 프레그먼트로 이동한다




